File: src/main/java/kr/co/bookand/backend/article/repository/ArticleRepository.java
package kr.co.bookand.backend.article.repository;

import kr.co.bookand.backend.article.domain.Article;
import kr.co.bookand.backend.article.domain.ArticleCategory;
import kr.co.bookand.backend.common.domain.Status;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ArticleRepository extends JpaRepository<Article, Long>, ArticleRepositoryCustom {
    boolean existsByTitle(String title);

    Long countAllByVisibility(boolean visibility);

    Long countAllByVisibilityAndCategory(boolean visibility, ArticleCategory articleCategory);

    Optional<Article> findFirstByStatusOrderByCreatedAtDesc(Status status);

    Long countAllByStatus(Status status);
}

File: src/main/kotlin/kr/co/bookand/backend/article/repository/ArticleRepository.kt
package kr.co.bookand.backend.article.repository

import kr.co.bookand.backend.article.model.Article
import kr.co.bookand.backend.article.model.ArticleCategory
import kr.co.bookand.backend.common.Status
import org.springframework.data.jpa.repository.JpaRepository

interface ArticleRepository : JpaRepository<Article, Long>, ArticleRepositoryCustom {
    fun existsByTitle(title: String): Boolean
    fun findFirstByStatusOrderByCreatedAtDesc(status: Status): Article?
    fun countAllByVisibility(visibility: Boolean): Long
    fun countAllByVisibilityAndCategory(visibility: Boolean, articleCategory: ArticleCategory): Long
    fun countAllByStatus(status: Status): Long
}